### PR TITLE
To be consistent with global flags in --logging-format description

### PR DESCRIPTION
**Pull request by chymy**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by chymy** [comment]
Please take a look, Thanks.
/assign @dchen1107 @yifan-gu @andrewsykim

File: cmd/kubelet/app/options/options.go [hunk 1]
```diff
@@ -544,7 +544,7 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
...
 	fs.StringVar(&c.KubeReservedCgroup, "kube-reserved-cgroup", c.KubeReservedCgroup, "Absolute name of the top level cgroup that is used to manage kubernetes components for which compute resources were reserved via '--kube-reserved' flag. Ex. '/kube-reserved'. [default='']")
-	fs.StringVar(&c.Logging.Format, "logging-format", c.Logging.Format, `Sets the log format. Permitted formats: "text", "json".\nNon-default formats don't honor these flags: -add_dir_header, --alsologtostderr, --log_backtrace_at, --log_dir, --log_file, --log_file_max_size, --logtostderr, --skip_headers, --skip_log_headers, --stderrthreshold, --log-flush-frequency.\nNon-default choices are currently alpha and subject to change without warning.`)
+	fs.StringVar(&c.Logging.Format, "logging-format", c.Logging.Format, `Sets the log format. Permitted formats: "text", "json". Non-default formats don't honor these flags: --add-dir-header, --alsologtostderr, --log-backtrace-at, --log-dir, --log-file, --log-file-max-size, --logtostderr, --skip-headers, --skip-log-headers, --stderrthreshold, --log-flush-frequency. Non-default choices are currently alpha and subject to change without warning.`)
```
Comment (ehashman) [inline cmd/kubelet/app/options/options.go:547]: You've removed the newlines here, I don't think we should do that.

**Review by ehashman — COMMENTED** [review]
/priority backlog
/triage accepted
/hold

File: cmd/kubelet/app/options/options.go (hunk 1 above)
Comment (chymy) [reply cmd/kubelet/app/options/options.go:547]: Thank you @ehashman . ``  will result in invalid newlines:
```
 Sets the log format. Permitted formats: "text", "json".\nNon-default formats don't honor these flags: -add_dir_header, --alsologtostderr, --log_backtrace_at, --log_dir, --log_file, --log_file_max_size, --logtostderr, --skip_headers, --skip_log_headers, --stderrthreshold, --log-flush-frequency.\nNon-default choices are currently alpha and subject to change without warning.
```

should use "". Fixed. Please take a look, thanks.

**Review by ehashman — COMMENTED** [review]
Hi @chymy,

I took a look and these flags appear to be directly passed to klog. The comment as written is accurate, you can see they use underscores, so this PR doesn't actually fix a typo:

https://github.com/kubernetes/kubernetes/blob/64a6ee4321d37740b84a5c5c7aaa67e0bf6b6508/vendor/k8s.io/klog/v2/klog.go#L434-L437

Hence, I will close this PR.

/close

**Comment by chymy** [comment]
Thanks @ehashman .
I change the title.  To be consistent with global flags in --logging-format description
global flags:
```
Global flags:

      --add-dir-header
                If true, adds the file directory to the header of the log messages
      --alsologtostderr
                log to standard error as well as files
  -h, --help
                help for kube-scheduler
      --log-backtrace-at traceLocation
                when logging hits line file:N, emit a stack trace (default :0)
      --log-dir string
                If non-empty, write log files in this directory
      --log-file string
                If non-empty, use this log file
      --log-file-max-size uint
                Defines the maximum size a log file can grow to. Unit is megabytes. If the value is 0, the maximum file size is
                unlimited. (default 1800)
      --log-flush-frequency duration
                Maximum number of seconds between log flushes (default 5s)
      --logtostderr
                log to standard error instead of files (default true)
      --one-output
                If true, only write logs to their native severity level (vs also writing to each lower severity level
      --skip-headers
                If true, avoid header prefixes in the log messages
      --skip-log-headers
                If true, avoid headers when opening log files
```
So in order to prevent readers from being confused, in --logging-format description should be:
```
--logging-format string
                Sets the log format. Permitted formats: "json", "text".
                Non-default formats don't honor these flags: --add-dir-header, --alsologtostderr, --log-backtrace-at, --log-dir,
                --log-file, --log-file-max-size, --logtostderr, --one-output, --skip-headers, --skip-log-headers, --stderrthreshold,
                --vmodule, --log-flush-frequency.
```
Would it be better?

/reopen

**Comment by ehashman** [comment]
@chymy can you please link the code that the output block you pasted originates from? I could not find these flags anywhere in the k8s code when I did a search, only the versions with underscores. Hence, I don't think this fixes anything.

**Comment by chymy** [comment]
> @chymy can you please link the code that the output block you pasted originates from? I could not find these flags anywhere in the k8s code when I did a search, only the versions with underscores. Hence, I don't think this fixes anything.

The underlined flags are in Klog and are normalized in k8s.
https://github.com/kubernetes/kubernetes/blob/e161051c8366c6c90502f1a4ef06da2f01db2264/staging/src/k8s.io/component-base/cli/globalflag/globalflags.go#L52-L54

So, in klog the flags :
```
--add_dir_header, --log_backtrace_at, --log_dir,
--log_file, --log_file_max_size,  --one_output, --skip_headers, --skip_log_headers
```

will be normalized to 
```
--add-dir_header,  --log-backtrace-at, --log-dir,
--log-file, --log-file-max-size, --one-output, --skip-headers, --skip-log-headers
```
As you can see in k8s：
```
Global flags:

      --add-dir-header
                If true, adds the file directory to the header of the log messages
      --alsologtostderr
                log to standard error as well as files
  -h, --help
                help for kube-scheduler
      --log-backtrace-at traceLocation
                when logging hits line file:N, emit a stack trace (default :0)
      --log-dir string
                If non-empty, write log files in this directory
      --log-file string
                If non-empty, use this log file
      --log-file-max-size uint
                Defines the maximum size a log file can grow to. Unit is megabytes. If the value is 0, the maximum file size is
                unlimited. (default 1800)
      --log-flush-frequency duration
                Maximum number of seconds between log flushes (default 5s)
      --logtostderr
                log to standard error instead of files (default true)
      --one-output
                If true, only write logs to their native severity level (vs also writing to each lower severity level
      --skip-headers
                If true, avoid header prefixes in the log messages
      --skip-log-headers
                If true, avoid headers when opening log files
```

So, in order to prevent the reader from making mistakes by using underlined flags, it is better to replace `_` with `-` in --logging-format description

eg:
```sh
# kube-controller-manager --add_dir_header
Error: unknown flag: --add_dir_header
```

**Review by ehashman — COMMENTED** [review]
Tested, newlines are now working.

/lgtm
/hold cancel

**Review by ehashman — COMMENTED** [review]
re
/lgtm

**Pull request merged**